framework/aster-frame/src/sync/rcu/mod.rs

```diff
@@ -70,35 +159,49 @@ pub struct RcuReclaimer<P> {
 impl<P: Send + 'static> RcuReclaimer<P> {
     pub fn delay(mut self) {
         let ptr: P = unsafe {
-            let ptr = core::mem::replace(&mut self.ptr, core::mem::uninitialized());
+            // SAFETY. It is ok to have this uninitialized value because
+            // 1) Its memory won't be acccessed;
+            // 2) It will be forgotten rather than being dropped;
+            // 3) Before it gets forgtten, the code won't return prematurely or panic.
+            #[allow(clippy::mem_replace_with_uninit, clippy::uninit_assumed_init)]
+            let ptr = core::mem::replace(
+                &mut self.ptr,
+                core::mem::MaybeUninit::uninit().assume_init(),
+            );
 
             core::mem::forget(self);
 
             ptr
         };
-        get_singleton().after_grace_period(move || {
-            drop(ptr);
-        });
-    }
-}
 
-impl<P> Drop for RcuReclaimer<P> {
```

Comment:
What happens if someone forgets to call `RcuReclaimer::leak`? Won't he immediately drop the RCU-protected data, violating the basic semantics of the RCU?

Reply:
Yes, you are right. I miss the situation that someone forgets to call `RcuReclaimer::delay`. I will implement `Drop` trait for `RcuReclaimer`. Thanks for your review!